src/ConnectedMode/Suppressions/ServerIssuesStore.cs

```diff
@@ -19,24 +19,59 @@
  */
 
 using System;
-using System.Linq;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.Linq;
 using SonarQube.Client.Models;
 
 namespace SonarLint.VisualStudio.ConnectedMode.Suppressions
 {
     [Export(typeof(IServerIssuesStore))]
+    [Export(typeof(IServerIssuesStoreWriter))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class ServerIssuesStore : IServerIssuesStore
+    internal class ServerIssuesStore : IServerIssuesStore, IServerIssuesStoreWriter
     {
-        private IEnumerable<SonarQubeIssue> serverIssues;
+        private List<SonarQubeIssue> serverIssues;
 
         public event EventHandler ServerIssuesChanged;
 
+        #region IServerIssuesStore implementation
+
         public IEnumerable<SonarQubeIssue> Get()
         {
             return serverIssues ?? Enumerable.Empty<SonarQubeIssue>();
         }
+
+        #endregion IServerIssuesStore implementation
+
+        #region IServerIssueStoreWriter implementation
+
+        public void AddIssues(IEnumerable<SonarQubeIssue> issues, bool clearAllExistingIssues)
```

Comment:
Yes, I think we do need to handle duplicates, unfortunately.
If two issues have the same `issueKey` we should consider them to be them same. We might need to implement a custom `IEqualityComparer<SonarQubeIssue>`, which we can then use to handle the differences.

I suggest handling duplicates in a separate PR. Let me know if you'd like to discuss implementation options.

Reply:
Does this mean I can assume in the UpdateIssue method that there will never be any duplicates?

Reply:
Yes, I think so. As it is written at the moment, it will throw if there are duplicates since you are using `SingleOrDefault` (`FirstOrDefault` doesn't throw if there are multiples).